src/actions/deepLinkActions.js

```diff
@@ -43,45 +78,12 @@ export const executeDeepLinkAction = (deepLink: string) => {
         }
         break;
       case 'approve':
-        const { query: { loginToken: loginAttemptToken } } = params;
-        dispatch({
-          type: ADD_DEEP_LINK_DATA,
-          payload: { loginAttemptToken },
-        });
-        const pathAndParams = getNavigationPathAndParamsState();
-        if (!pathAndParams) {
-          updateNavigationLastScreenState({
-            lastActiveScreen: HOME,
-            lastActiveScreenParams: {
-              showLoginApproveModal: true,
-            },
-          });
-          break;
-        }
-        const pathParts = pathAndParams.path.split('/');
-        const currentFlow = pathParts[0];
-        const currentScreen = pathParts[pathAndParams.path.length - 1];
-        if (currentScreen !== HOME) {
-          updateNavigationLastScreenState({
-            lastActiveScreen: HOME,
-            lastActiveScreenParams: {
-              showLoginApproveModal: true,
-            },
-          });
-          if (currentFlow !== AUTH_FLOW) {
-            const navigateToAppAction = NavigationActions.navigate({
-              routeName: APP_FLOW,
-              params: {},
-              action: NavigationActions.navigate({
-                routeName: HOME,
-                params: {
-                  showLoginApproveModal: true,
-                },
-              }),
-            });
-            navigate(navigateToAppAction);
-          }
-        }
+        const {
+          query = { loginToken: '' },
```

Comment:
I'm sorry, I think default here is useless, you can just check in `beginApproveLogin` if loginToken is present or not.

Reply:
Yes, the thing is that the `loginToken` field is mandatory on the `ApproveLoginQuery` type, since I thought it wouldn't make sense to call `beginApproveLogin` without a token.

I think you're right, tho, we could make that field optional and let `beginApproveLogin` handle the rest.